src/engine/OptionalJoin.cpp

```diff
@@ -287,6 +287,16 @@ auto OptionalJoin::computeImplementationFromIdTables(
   return implementation;
 }
 
+// _____________________________________________________________________________
+bool OptionalJoin::columnOriginatesFromGraph(const Variable& variable) const {
+  AD_CONTRACT_CHECK(getExternallyVisibleVariableColumns().contains(variable));
+  if (_left->getVariableColumnOrNullopt(variable).has_value() &&
+      _right->getVariableColumnOrNullopt(variable).has_value()) {
+    return _left->getRootOperation()->columnOriginatesFromGraph(variable);
```

Comment:
Add a comment here, this is subtle, and I first thought that it was wrong, but it probably is correct.

In particular comment, that this is the behavior from the join columns, which will always be unchanged from the left input.